deployer/models/release_safe.go

```diff
@@ -60,122 +60,194 @@ func (release *Release) ValidateSafeRelease(s3c aws.S3API, resources *ReleaseRes
+		errstr = appendError(errstr, srse.TargetGroups)
+		errstr = appendError(errstr, srse.EBSVolumeSize)
+		errstr = appendError(errstr, srse.EBSVolumeType)
+		errstr = appendError(errstr, srse.EBSDeviceName)
+		errstr = appendError(errstr, srse.AssociatePublicIpAddress)
+		errstr = appendError(errstr, srse.InstanceType)
+		errstr = appendError(errstr, srse.MinSize)
+		errstr = appendError(errstr, srse.MaxSize)
+		errstr = appendError(errstr, srse.MaxTerminations)
+		errstr = appendError(errstr, srse.DefaultCooldown)
+		errstr = appendError(errstr, srse.HealthCheckGracePeriod)
+		errstr = appendError(errstr, srse.Spread)
+	}
+
+	return errstr
+}
+
+func appendError(errstr string, err error) string {
+	if err != nil {
+		errstr = fmt.Sprintf("%s\n%s", errstr, err.Error())
+	}
+
+	return errstr
+}
+
+// Returns nil if there is no error contained
+func (sre *SafeReleaseError) IsError() error {
+	if sre.Error() == "" {
+		return nil
+	}
+
+	return sre
+}
+
 func (release *Release) validateSafeRelease(previousRelease *Release) error {
+	sre := &SafeReleaseError{
+		Services: map[string]*SafeReleaseServiceError{},
+	}
+
 	// 1. Subnets, or Services
 	if res := safeUnorderedStrList(release.Subnets, previousRelease.Subnets); res != nil {
-		return fmt.Errorf("SafeRelease Error: Subnets different %v", *res)
+		sre.Subnets = fmt.Errorf("SafeRelease Error: Subnets different %v", *res)
 	}
 
 	if res := safeInt(release.Timeout, previousRelease.Timeout); res != nil {
-		return fmt.Errorf("SafeRelease Error: Timeout different %v", *res)
+		sre.Timeout = fmt.Errorf("SafeRelease Error: Timeout different %v", *res)
 	}
 
-	if err := validateSafeServices(release.Services, previousRelease.Services); err != nil {
-		return err
-	}
+	// This will add errors to the sre
+	validateSafeServices(sre, release.Services, previousRelease.Services)
 
-	return nil
+	// Check whether an error was found and return if it has
+	return sre.IsError()
 }
 
-func validateSafeServices(services map[string]*Service, prevServices map[string]*Service) error {
+func validateSafeServices(sre *SafeReleaseError, services map[string]*Service, prevServices map[string]*Service) {
 	if res := safeUnorderedStrList(serviceMapKeys(services), serviceMapKeys(prevServices)); res != nil {
-		return fmt.Errorf("SafeRelease Error: Incorrect Services service %v", *res)
+		sre.AllServices = fmt.Errorf("SafeRelease Error: Incorrect Services service %v", *res)
 	}
 
 	for serviceName, service := range services {
 		prevService, ok := prevServices[serviceName]
 
 		if !ok {
-			return fmt.Errorf("SafeRelease Error(%v): No previous service", serviceName)
+			// Pretty sure this will never be reached (best check though)
+			sre.MissingService = fmt.Errorf("SafeRelease Error(%v): No previous service", serviceName)
 		}
 
-		if err := validaeSafeService(serviceName, service, prevService); err != nil {
-			return err
-		}
+		sre.Services[serviceName] = validaeSafeService(serviceName, service, prevService)
```

Comment:
typo, validae => validate

Reply:
you don't know that, maybe I meant validae :) I will fix